source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -521,11 +521,12 @@ class ComponentInterface {
   /**
    * @brief Validate an add_service request by parsing the service name and checking the maps of registered services.
    * @param service_name The name of the service
+   * @param type One of empty|string
    * @throws modulo_utils::exceptions::AddServiceException if the service could not be created
    * (empty name or already registered)
    * @return The parsed service name
    */
-  std::string validate_service_name(const std::string& service_name);
+  std::string validate_service_name(const std::string& service_name, const std::string& type) const;
```

Comment:
this is only called internally so it's not critical, but when a string input is restricted to specific constants then it should rather be an enumeration.